src/docker/install.ts

```diff
@@ -376,8 +381,10 @@ export class Install {
           await Exec.exec('sudo', ['sh', '-c', 'echo 0 > /proc/sys/kernel/apparmor_restrict_unprivileged_userns']);
         }
       }
-
-      const cmd = `${dockerPath} --host="${dockerHost}" --config-file="${daemonConfigPath}" --exec-root="${this.runDir}/execroot" --data-root="${this.runDir}/data" --pidfile="${this.runDir}/docker.pid"`;
+      let cmd = `${dockerPath} --host="${dockerHost}" --config-file="${daemonConfigPath}" --exec-root="${this.runDir}/execroot" --data-root="${this.runDir}/data" --pidfile="${this.runDir}/docker.pid"`;
+      if (this.localTCPPort) {
+        cmd += ` --host="tcp://127.0.0.1:${this.localTCPPort}"`;
```

Comment:
Note: Exposing the docker socket over TCP without TLS is deprecated and will be a hard failure in future:

```
WARN[2024-12-13T11:24:36.737970212Z] Binding to IP address without --tlsverify is insecure and gives root access on this machine to everyone who has access to your network.  host="tcp://127.0.0.1:1234"
WARN[2024-12-13T11:24:36.738007253Z] Binding to an IP address, even on localhost, can also give access to scripts run in a browser. Be safe out there!  host="tcp://127.0.0.1:1234"
WARN[2024-12-13T11:24:36.738011878Z] [DEPRECATION NOTICE] In future versions this will be a hard failure preventing the daemon from starting! Learn more at: https://docs.docker.com/go/api-security/  host="tcp://127.0.0.1:1234"
```

See: https://github.com/moby/moby/pull/41285

Reply:
I do agree but hope there will be some opt-in env var to still avoid TLS for development purpose.